internal/states/encryption/instance.go

```diff
@@ -0,0 +1,119 @@
+package encryption
+
+import (
+	"strings"
+
+	"github.com/opentofu/opentofu/internal/logging"
+	"github.com/opentofu/opentofu/internal/states/encryption/encryptionconfig"
+	"github.com/opentofu/opentofu/internal/states/encryption/encryptionflow"
+)
+
+// --------------------------------------------------------------------------------------------
+// IMPORTANT NOTE:
+// This package contains a cache for singleton instances of encryptionflow.FlowBuilder
+//   - there is one instance for our own remote state
+//   - there is one instance for our local state file
+//   - there is one instance for our local plan file
+//   - there is one instance for each remote state data source
+//
+// One way to configure state/plan encryption is adding settings to the terraform {} block
+// (or the terraform_remote_state block, respectively). These blocks are parsed by code far removed
+// from where they are used, and they are parsed multiple times by OpenTofu.
+//
+// Combine that with procedural code in internal/states/statefile (local state file) and
+// internal/plans (local plan file), which is called from all over the place.
+//
+// This is why caching singleton instances in this package is the less painful option.
```

Comment:
Whilst I understand it's parsed multiple times and in multiple places and that's why you wanted to have the singleton, is it worth the technical debt of this layer to introduce a singleton/cache/etc instead of parsing it each time?

I would argue that we're maybe over complicating things here with this layer.